src/api/interface.ts

```diff
@@ -21,24 +21,50 @@ enum ResolveError {
 /**
 * Intent descriptor
 */
-interface IntentMetadata {
+interface Intent {
+  
+  // The name of the intent, default naming convention is UpperCamelCase
   name: string;
-  displayName: string;
+
+  // An optional display name for the intent, "name" will be used if left empty
+  displayName?: string;
+  
+  // A list of the contexts the intent accepts. This will typically be a set of namespaced context types, e.g. "fdc3.contact"
+  contexts?: string[]
+ 
+  // Custom configuration for the intent that may be required for a particular desktop agent.
+  customConfig?: object; 
 }
 
 /**
 * An interface that relates an intent to apps
 */
 interface AppIntent {
-  intent: IntentMetadata;
-  apps: Array<AppMetadata>;
+  intent: Intent;
+  apps: Array<Application>;
 }
 
 /**
- * App metadata is Desktop Agent specific - but should support a name property.
+ * App definition as provided by the application directory
  */
-interface AppMetadata {
+interface Application {
+  appId: string;
   name: string;
+  title?: string;
+  tooltip?: string;
+  description?: string;
+  images?: Array<AppImage>;
+  icons?: Array<Icon>;
+  intents?: Array<Intent>;
+  customConfig?: Array<{name: string, value: string}>;
+}
```

Comment:
Happy to exclude some of these properties, or include some more - it doesn't have to be an exact 1-to-1 match with the AppD schema, only metadata that will be useful for applications using the APIs.

For example, here I have left out `manifest` and `manifestType` (which are mandatory in the AppD schema), as I don't see what use they would have outside of the App Directory. But I am not against including them as well for consistency...